master/contrib/github_buildbot.py

```diff
@@ -153,28 +174,68 @@ def process_change(self, payload, user, repo, repo_url, project, request):
                 Hook.
         """
         changes = None
-        branch = payload['ref'].split('/')[-1]
+        refname = payload['ref']
+
+        m = re.match(r"^refs\/(heads|tags)\/(.+)$", refname)
+        if not m:
+            logging.info("Ignoring refname `%s': Not a branch or tag", refname)
+            return changes
+
+        branch = m.group(2)
```

Comment:
this is misleading in the light of the above regular expression: it's either a branch or a tag.